tests/compile-fail/invalid_upcast_comparisons.rs

```diff
@@ -0,0 +1,19 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(invalid_upcast_comparisons)]
+#![allow(unused, eq_op, no_effect)]
+fn main() {
+    let zero: u32 = 0;
+    let u8_max: u8 = 255;
+
+    (u8_max as u32) > 300; //~ERROR because of the numeric bounds on `u8_max` prior to casting, this expression is always false
+    (u8_max as u32) > 20;
+
+    (zero as i32) < -5; //~ERROR because of the numeric bounds on `zero` prior to casting, this expression is always false
+    (zero as i32) < 10;
+
+    -5 < (zero as i32); //~ERROR because of the numeric bounds on `zero` prior to casting, this expression is always true
+    0 <= (zero as i32); //~ERROR because of the numeric bounds on `zero` prior to casting, this expression is always true
+    0 < (zero as i32);
+}
```

Comment:
You have tests with `lit < casted_expr` and `casted_expr > lit` which are basically the same thing thanks to the `normalize_comparison` function. I’d like to see another test with `lit > casted_expr`. The lint also makes sense for `==` and `!=`:
```rust
(u8_max as u32) == 1337; // always false
```